src/Umbraco.Infrastructure/Migrations/Install/DatabaseDataCreator.cs

```diff
@@ -1745,14 +1747,104 @@ private void CreatePropertyTypeData()
         }
     }
 
-    private void CreateLanguageData() =>
-        ConditionalInsert(
+    private void CreateLanguageData()
+    {
+        // For languages we support the installation of records that are additional to the default installed data.
+        // We can do this as they are specified by ISO code, which is enough to fully detail them.
+        // All other customizable install data is specified by GUID, and hence we only know about the set that are installed by default.
+        InstallDefaultDataSettings? languageInstallDefaultDataSettings = _installDefaultDataSettings.Get(Constants.Configuration.NamedOptions.InstallDefaultData.Languages);
+        if (languageInstallDefaultDataSettings?.InstallData == InstallDefaultDataOption.Values)
+        {
+            // Insert the specified languages, ensuring the first is marked as default.
+            bool isDefault = true;
+            foreach (var isoCode in languageInstallDefaultDataSettings.Values)
+            {
+                if (!TryCreateCulture(isoCode, out CultureInfo? culture))
+                {
+                    continue;
+                }
+
+                var dto = new LanguageDto
+                {
+                    IsoCode = culture.Name,
+                    CultureName = culture.DisplayName,
```

Comment:
I just noticed this should've used `EnglishName` instead:
```suggestion
                    CultureName = culture.EnglishName,
```